### ISSUE TITLE
nerd fonts aren't rendering as expected in the web view

### ISSUE THREAD

**Issue by fdncred** [rgwood/escape-artist#8]
As we discussed on Discord. There are glyphs that aren't appearing in the webview. You can see them in the second screenshot as little squares. I also noticed that the `+5`, `main` and the time are rendered with a different foreground color in the webview.

My prompt.
![image](https://github.com/rgwood/escape-artist/assets/343840/a1f1a558-e26e-48ac-9873-284da6e5f66d)
How it's rendering in MS Edge on Windows.
![image](https://github.com/rgwood/escape-artist/assets/343840/61cf4ded-d999-43d1-b08b-3001db043717)

**Comment by rgwood** [comment]
Good catch! Should be fixed by https://github.com/rgwood/escape-artist/commit/07f1271ddfc14c146ccebbde249ccc297c913ab3 , available now in v0.6.7

**Comment by fdncred** [comment]
Looks much better now.
![image](https://github.com/rgwood/escape-artist/assets/343840/0638d54e-dbad-4561-90d6-202892c750c0)

The `main` and `+5` are the wrong foreground color/text color but the nerd fonts look great! Thanks for fixing it.

**Comment by rgwood** [comment]
I suspect what’s going on there is that your prompt is using 4-bit colours and those are not super well defined; Escape Artist doesn’t know exactly what colours your terminal will use so it has to guess (IIRC the library I use uses the default colour palette from xterm).

**Comment by fdncred** [comment]
In the git_segment `$TERM_BG` is set as the fg color and it is color `"#0C0C0C"` and bg color is `"4E9A06"`, which it seems to get the bg right.
https://github.com/nushell/nu_scripts/blob/main/modules/prompt/oh-my.nu#L346

Here repo_status is a bit harder to read but these are the colors it uses for fg and bg. The bg is right on this one too.
https://github.com/nushell/nu_scripts/blob/main/modules/prompt/oh-my.nu#L222-L223
```nushell
    let GIT_BG = "#C4A000"
    let GIT_FG = "#000000"
```

Are #0c0c0c and #000000 considered 4-bit colors here?

**Comment by rgwood** [comment]
I find it hard to diagnose this based on your prompt config; I think it would be more useful to see the escape codes that are setting the foreground+background colour.

**Comment by fdncred** [comment]
Oh, good idea. It can be hard to understand after so many translations. Sorry for not leading with this file. Still wrapping my head around Escape Artist's full functionality.
[stdout.txt](https://github.com/rgwood/escape-artist/files/14946888/stdout.txt)

It looks to me like the branch is using escape 38;2 for fg and 48;2 for bg, both which pass a rgb colors. Which, in my mind, is what should be happening considering our nushell ansi code interprets hex colors as rgb.

**Comment by rgwood** [comment]
Hmm, that is odd. I can't reproduce the foreground colour issue with `$"(ansi { fg: '#0C0C0C' bg: '#4E9A06'})) foo (ansi reset)"` in Nu, and from a quick glance at your recording the foreground colour in your screenshot does seem wrong.

I should probably build out functionality to replay the `stdout.txt` recordings but I don't think I'll have time anytime soon.

**Comment by fdncred** [comment]
No worries at all. I appreciate all your help.

**Comment by rgwood** [comment]
I lied, I had more energy after dinner and I'm now pretty far along with the implementation of the replay functionality.

Interestingly, I can't reproduce the bug in your screenshot; the foreground text looks correct now

![image](https://github.com/rgwood/escape-artist/assets/26268125/698fca72-2ec7-4120-87fe-0ace4059f478)

**Comment by rgwood** [comment]
I'm also noticing some odd behaviour on Windows. It's almost like Nu tries to take over the entire terminal when running on Windows under Escape Artist; it erases the entire screen before showing the elephant. It doesn't do this when launched outside of Escape Artist... I wonder if Escape Artist accidentally triggers a weird code path in Reedline or Crossterm?

![image](https://github.com/rgwood/escape-artist/assets/26268125/4907e795-242e-4b6e-8663-44105cf6f928)

**Comment by fdncred** [comment]
I figured it out and it's all my fault. I'll document it here so if someone else is as stupid as I am, they can come here and see what's going on.

Here's what was happening. I started debugging again today and decided to look at the browswer dev tools. This is what I saw. The dev tools is for the hamburger icon, the hover is just showing what color escape-artist thinks it is.
![image](https://github.com/rgwood/escape-artist/assets/343840/942fe957-4c00-492a-ace1-ae4cea22f322)

When I saw `rgb(0, 0, 0)` I knew that the color was right and escape-artist was not breaking things. Then I read more and saw `darkreader` and immediately knew what was going on. 

I need to turn this thing off for localhost.
![image](https://github.com/rgwood/escape-artist/assets/343840/b530784c-0151-4613-a33e-1d784f635e56)

The same line now looks like this.
![image](https://github.com/rgwood/escape-artist/assets/343840/5730e95b-c7d1-4ae6-a9ec-c4f0a00a69c6)

Sorry for bothering you with my mistake but at least I figured it out.
